services/repository/adopt.go

```
@@ -84,23 +90,18 @@ func AdoptRepository(ctx context.Context, doer, u *user_model.User, opts CreateR
 			return fmt.Errorf("checkDaemonExportOK: %w", err)
 		}
 
-		// Initialize Issue Labels if selected
-		if len(opts.IssueLabels) > 0 {
-			if err := repo_module.InitializeLabels(ctx, repo.ID, opts.IssueLabels, false); err != nil {
-				return fmt.Errorf("InitializeLabels: %w", err)
-			}
-		}
-
 		if stdout, _, err := git.NewCommand(ctx, "update-server-info").
 			RunStdString(&git.RunOpts{Dir: repoPath}); err != nil {
 			log.Error("CreateRepository(git update-server-info) in %v: Stdout: %s\nError: %v", repo, stdout, err)
 			return fmt.Errorf("CreateRepository(git update-server-info): %w", err)
 		}
 		return nil
-	}); err != nil {
+	}(); err != nil {
+		if errDel := DeleteRepository(ctx, doer, repo, false /* no notify */); errDel != nil {
+			log.Error("Failed to delete adopt-failed repository %s: %v", repo.FullName(), errDel)
```

```suggestion
			log.Error("Failed to delete repository %s that could not be adopted (original error: %v): %v", repo.FullName(), err, errDel)
```

---

I think the rule is:

* If an error is returned, then it doesn't need to be outputed to log, it should be handled by parent callers.
* If an error is not returned, then it's better to output it into log.

This line is the first case: `err` will be returned and outputed by caller. So here we doesn't need to print it again, otherwise, duplicate.

---

```suggestion
			log.Error("Failed to delete repository %s that could not be adopted: %v", repo.FullName(), errDel)
```